### PR TITLE
[libc++] Add a script to synchronize status-tracking CSVs with Github issues

### PR DESCRIPTION
**Pull request by ldionne**
This script can be run manually to synchronize the CSV files that we use to track Standards Conformance with the Github issues that track our implementation of LWG issues and papers.

### CODE REVIEW

**Comment by h-vetinari** [comment]
This is cool! :)

For all the "non-special" ones (i.e. `issue['status'] == ''`), it might be worth a thought to insert a link back to the github issue into the status page. I was even thinking` about doing this manually (because IMO it's a value add to be able to go from the status page to the issue where the approach for an implementation is discussed), but if this becomes scriptable, it'd be even easier. The sync logic would have to become a bit smarter, but nothing extraordinary.

File: libcxx/utils/synchronize_csv_status_files.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+import csv
+import itertools
+import json
+import os
+import pathlib
+import re
+import subprocess
+
+# Number of the 'Libc++ Standards Conformance' project on Github
+LIBCXX_CONFORMANCE_PROJECT = '31'
```
Comment (mordante) [inline libcxx/utils/synchronize_csv_status_files.py:13]: This sounds quite fragile, is there another way to implement this based on the labels in GitHub?

File: libcxx/utils/synchronize_csv_status_files.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
+#!/usr/bin/env python3
```
Comment (mordante) [inline libcxx/utils/synchronize_csv_status_files.py:1]: We should include a copyright header.

**Review by mordante — COMMENTED** [review]
> This is cool! :)
> 
> For all the "non-special" ones (i.e. `issue['status'] == ''`), it might be worth a thought to insert a link back to the github issue into the status page. I was even thinking` about doing this manually (because IMO it's a value add to be able to go from the status page to the issue where the approach for an implementation is discussed), but if this becomes scriptable, it'd be even easier. The sync logic would have to become a bit smarter, but nothing extraordinary.

IIRC this is something we discussed and in favour of.

@ldionne can you run this script and upload the changes to the CSV files? This makes it easier to review the changes.

File: libcxx/utils/synchronize_csv_status_files.py (hunk 1 above)
Comment (h-vetinari) [reply libcxx/utils/synchronize_csv_status_files.py:13]: Project numbers are as stable as issue numbers, so that shouldn't be an issue IMO.

Comment (ldionne) [reply libcxx/utils/synchronize_csv_status_files.py:13]: ^ This. I am not aware of any way to refer to a project by name, and given that this is basically like an issue number, I think it's the best way to refer to it. Although I agree that it looks weird at first.

**Comment by ldionne** [comment]
> 
> @ldionne can you run this script and upload the changes to the CSV files? This makes it easier to review the changes.

Running the script currently doesn't produce any diff. I made some NFC tweaks to our CSV files so they would be consistent to the point where running the script doesn't produce any difference.

**Comment by ldionne** [comment]
As discussed in the monthly meeting, merging.

**Pull request merged**